### ISSUE TITLE
won't build on manylinux2010_x86_64 docker image

### ISSUE THREAD

**Issue by chinmayanagpal** [libsemigroups/libsemigroups_pybind11#37]
Output:
```
  + python -m pip wheel /project --wheel-dir=/tmp/cibuildwheel/built_wheel --no-deps
    DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
     pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
    ERROR: Command errored out with exit status 1:
     command: /opt/python/cp36-cp36m/bin/python /opt/python/cp36-cp36m/lib/python3.6/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmp3py3dh_v
         cwd: /tmp/pip-req-build-ubqg6em2
    Complete output (46 lines):
    /opt/rh/devtoolset-8/root/usr/lib64:/opt/rh/devtoolset-8/root/usr/lib:/opt/rh/devtoolset-8/root/usr/lib64/dyninst:/opt/rh/devtoolset-8/root/usr/lib/dyninst:/usr/local/lib64:/usr/lib
    running bdist_wheel
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.6
    creating build/lib.linux-x86_64-3.6/libsemigroups_pybind11
    copying libsemigroups_pybind11/matrix.py -> build/lib.linux-x86_64-3.6/libsemigroups_pybind11
    copying libsemigroups_pybind11/__init__.py -> build/lib.linux-x86_64-3.6/libsemigroups_pybind11
    copying libsemigroups_pybind11/tools.py -> build/lib.linux-x86_64-3.6/libsemigroups_pybind11
    copying libsemigroups_pybind11/froidure_pin.py -> build/lib.linux-x86_64-3.6/libsemigroups_pybind11
    copying libsemigroups_pybind11/transf.py -> build/lib.linux-x86_64-3.6/libsemigroups_pybind11
    running build_ext
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/opt/python/cp36-cp36m/include/python3.6m -c flagcheck.cpp -o flagcheck.o -std=c++17
    building '_libsemigroups_pybind11' extension
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/src
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/usr/local/include -I/usr/local/include/libsemigroups -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/opt/python/cp36-cp36m/include/python3.6m -c src/forest.cpp -o build/temp.linux-x86_64-3.6/src/forest.o -std=c++17 -fvisibility=hidden -g0
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/usr/local/include -I/usr/local/include/libsemigroups -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/opt/python/cp36-cp36m/include/python3.6m -c src/words.cpp -o build/temp.linux-x86_64-3.6/src/words.o -std=c++17 -fvisibility=hidden -g0
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/usr/local/include -I/usr/local/include/libsemigroups -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/opt/python/cp36-cp36m/include/python3.6m -c src/bmat8.cpp -o build/temp.linux-x86_64-3.6/src/bmat8.o -std=c++17 -fvisibility=hidden -g0
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/usr/local/include -I/usr/local/include/libsemigroups -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/opt/python/cp36-cp36m/include/python3.6m -c src/todd-coxeter.cpp -o build/temp.linux-x86_64-3.6/src/todd-coxeter.o -std=c++17 -fvisibility=hidden -g0
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/usr/local/include -I/usr/local/include/libsemigroups -I/tmp/pip-build-env-ki6tcz0u/overlay/lib/python3.6/site-packages/pybind11/include -I/opt/python/cp36-cp36m/include/python3.6m -c src/matrix.cpp -o build/temp.linux-x86_64-3.6/src/matrix.o -std=c++17 -fvisibility=hidden -g0
    src/matrix.cpp: In lambda function:
    src/matrix.cpp:162:58: error: no matching function for call to ‘std::basic_string<char>::erase(__gnu_cxx::__normal_iterator<const char*, std::basic_string<char> >, std::basic_string<char>::const_iterator)’
                          str.erase(str.cend() - 3, str.cend());
                                                              ^
    In file included from /opt/rh/devtoolset-8/root/usr/include/c++/8/string:52,
                     from /opt/rh/devtoolset-8/root/usr/include/c++/8/stdexcept:39,
                     from /opt/rh/devtoolset-8/root/usr/include/c++/8/array:39,
                     from /opt/rh/devtoolset-8/root/usr/include/c++/8/tuple:39,
                     from /opt/rh/devtoolset-8/root/usr/include/c++/8/bits/unique_ptr.h:37,
                     from /opt/rh/devtoolset-8/root/usr/include/c++/8/memory:80,
                     from src/matrix.cpp:27:
    /opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h:4632:7: note: candidate: ‘std::basic_string<_CharT, _Traits, _Alloc>& std::basic_string<_CharT, _Traits, _Alloc>::erase(std::basic_string<_CharT, _Traits, _Alloc>::size_type, std::basic_string<_CharT, _Traits, _Alloc>::size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
           erase(size_type __pos = 0, size_type __n = npos)
           ^~~~~
    /opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h:4632:7: note:   no known conversion for argument 1 from ‘__gnu_cxx::__normal_iterator<const char*, std::basic_string<char> >’ to ‘std::basic_string<char>::size_type’ {aka ‘long unsigned int’}
    /opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h:4648:7: note: candidate: ‘std::basic_string<_CharT, _Traits, _Alloc>::iterator std::basic_string<_CharT, _Traits, _Alloc>::erase(std::basic_string<_CharT, _Traits, _Alloc>::iterator) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::basic_string<_CharT, _Traits, _Alloc>::iterator = __gnu_cxx::__normal_iterator<char*, std::basic_string<char> >; typename _Alloc::rebind<_CharT>::other::pointer = char*]’
           erase(iterator __position)
           ^~~~~
    /opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h:4648:7: note:   candidate expects 1 argument, 2 provided
    /opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h:4668:7: note: candidate: ‘std::basic_string<_CharT, _Traits, _Alloc>::iterator std::basic_string<_CharT, _Traits, _Alloc>::erase(std::basic_string<_CharT, _Traits, _Alloc>::iterator, std::basic_string<_CharT, _Traits, _Alloc>::iterator) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::basic_string<_CharT, _Traits, _Alloc>::iterator = __gnu_cxx::__normal_iterator<char*, std::basic_string<char> >; typename _Alloc::rebind<_CharT>::other::pointer = char*]’
           erase(iterator __first, iterator __last);
           ^~~~~
    /opt/rh/devtoolset-8/root/usr/include/c++/8/bits/basic_string.h:4668:7: note:   no known conversion for argument 1 from ‘__normal_iterator<const char*,[...]>’ to ‘__normal_iterator<char*,[...]>’
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
    ERROR: Failed building wheel for libsemigroups-pybind11
  ERROR: Failed to build one or more wheels
```

**Comment by james-d-mitchell** [comment]
Hmm, this looks like an issue in libsemigroups itself (albeit one that should be easy to fix). Which compiler is being used, and which version?

**Comment by james-d-mitchell** [comment]
> Hmm, this looks like an issue in libsemigroups itself (albeit one that should be easy to fix). Which compiler is being used, and which version?

I take it back, it's a bug in `libsemigroups_pybind11` can you double check if the fix in #38, actually resolves this or not?

**Comment by chinmayanagpal** [comment]
Resolved in #38
